src/Google.Api.Gax/CallSettings.cs

```diff
@@ -62,6 +73,65 @@ public sealed class CallSettings
             WriteOptions = WriteOptions,
             PropagationToken = PropagationToken,
             Credentials = Credentials,
+            RetrySettings = RetrySettings?.Clone(),
         };
+
+        // Merge other into this.
+        // Other is unchanged. Returns this
+        // Internal? Or Public?
+        public CallSettings Merge(CallSettings other)
+        {
+            if (other == null)
+            {
+                return this;
+            }
+            if (other.Headers != null)
+            {
+                Headers = other.Headers.Clone();
+            }
+            if (other.Expiration != null)
```

Comment:
All of the replace properties could be written as

```csharp
Expiration = other.Expiration ?? Expiration;
// etc
```

... which has a pointless assignment, but would be a lot shorter, and easier to see ones which weren't just replacements.

(Of course, if we make merge return a new instance instead, the assignment wouldn't be pointless...)